pipenv/cli/command.py

```diff
@@ -323,6 +323,15 @@ def lock(ctx, state, **kwargs):
     dev_only = state.lockoptions.dev_only
     pre = state.installstate.pre
     if emit_requirements:
+        echo(
+            crayons.yellow(
+                """
+            Warning: The lock flags -r/--requirements, along with it's options (--keep-outdated, --dev, --dev-only)
```

Comment:
I think you cannot deprecate `--keep-outdated` because it is used change the behavior of locking and thus install as well.   I think the `--dev` or `--dev-only` may also be used similarly .. isn't `--dev` how you specify to only install the development requirements?  I am not sure about `--dev-only` actually.

Otherwise I think this is the right direction to head in.

Unrelated, can you also have a look at this PR:  https://github.com/pypa/pipenv/pull/5071

Reply:
Oh one more thing, we are trying to not add any more crayons usage -- use click coloring instead.  @oz123 will thank you.  I have a recent example of doing just that:  https://github.com/pypa/pipenv/pull/5046/files#diff-ef852c4ac364f946819f765a6bc26f04f1b0968f31fc512949a60fa2ab0685e8R903-R910

Reply:
Does `requirements` subcommand updates the packages? Is the default `keep-outdated` or update?
This should be clarified in the deperecation message.

Reply:
I've updated the message. It indeed was very confusing. Apologies. Please check if it is okay now @matteius @oz123